app/controllers/sog/name-trainer.js

```diff
@@ -19,131 +22,164 @@ export default Controller.extend({
+  }
+
+  get progress() {
     return (this.currentQuestionIndex / this.questions.length) * 100;
-  }),
-  inputClass: computed('answered', 'currentQuestion.success', function () {
+  }
+
+  get inputClass() {
     if (this.answered) {
       if (this.currentQuestion.success) {
         return 'form-control is-valid';
       } else {
         return 'form-control is-invalid';
       }
     }
-
     return 'form-control';
-  }),
-  grade: computed('questions.length', 'success', function () {
+  }
+
+  get grade() {
     return 1 + Math.round((90 / this.questions.length) * this.success) / 10;
-  }),
-  actions: {
-    setGroupId(model) {
-      this.set('groupId', model.id);
-    },
-    startTrainer() {
-      this.users.then((users) => {
-        this.generateQuestions(users.filter((user) => user.avatarThumbUrl));
-        this.set('started', true);
-        this.set('finished', false);
-        this.set('currentQuestionIndex', 1);
-        this.set('success', 0);
-      });
-    },
-    stopTrainer() {
-      this.set('started', false);
-      this.set('finished', false);
-    },
-    chooseOption(option) {
-      if (!this.answered) {
-        let { currentQuestion } = this;
-        currentQuestion.answer = option;
-        currentQuestion.success = option === currentQuestion.question;
-        if (currentQuestion.success) {
-          this.incrementProperty('success');
-        }
-
-        this.goToNextQuestion();
-      }
-    },
-    checkAnswer() {
+  }
+
+  @action
+  setGroupId(model) {
+    this.groupId = model.id;
+  }
+
+  @action
+  async startTrainer() {
+    const users = await this.users;
+    try {
+      this.generateQuestions(users.filter((user) => user.avatarThumbUrl));
+      this.started = true;
+      this.finished = false;
+      this.currentQuestionIndex = 1;
+      this.success = 0;
+    } catch (error) {
+      this.errorMessage = error.message;
+    }
+  }
+
+  @action
+  stopTrainer() {
+    this.started = false;
+    this.finished = false;
+  }
+
+  @action
+  chooseOption(option) {
+    if (!this.answered) {
       let { currentQuestion } = this;
-      currentQuestion.answer = this.textInput;
-      currentQuestion.success =
-        currentQuestion.answer === currentQuestion.question.get('fullName');
+      currentQuestion.answer = option;
+      currentQuestion.success = option === currentQuestion.question;
       if (currentQuestion.success) {
-        this.incrementProperty('success');
+        this.success++;
       }
-
       this.goToNextQuestion();
-    },
-  },
+    }
+  }
+
+  @action
+  checkAnswer() {
+    let { currentQuestion } = this;
+    currentQuestion.answer = this.textInput;
+    currentQuestion.success =
+      currentQuestion.answer === currentQuestion.question.get('fullName');
+    if (currentQuestion.success) {
+      this.success++;
+    }
+    this.goToNextQuestion();
+  }
+
   generateQuestions(people) {
+    if (!people.length) {
```

Comment:
I don't like this control flow and think you should rather check people.length in the startTrainer action and set errorMessage there directly, as generating questions for no people should logically return no questions instead of an error

Reply:
good point, I think was a bit tired when I wrote this.

Reply:
fixed, tell me if this is what you had in mind